src/components/side_panel/pivot/pivot_layout_configurator/pivot_dimension/pivot_dimension.xml

```diff
@@ -16,11 +16,20 @@
           />
           <span t-else="1" class="fw-bold" t-esc="props.dimension.displayName"/>
         </div>
-        <i
-          class="btn fa fa-trash o-text-muted pe-0 ms-auto"
-          t-if="props.onRemoved"
-          t-on-click="() => props.onRemoved(props.dimension)"
-        />
+        <div class="d-flex flex-row align-items-center">
+          <i
```

Comment:
in the calculated measure PR, I added a slot `<t t-slot="upper-right-icons"/>` exactly here for the exact same reason (add a button icon + associated behavior)

I thing the slot idea is better since this component can remain generic and agnostic to the type of dimension.
I also extracted all the measure related stuff in its own component to help keeping things tidy